scripts/make_descs.py

```diff
@@ -0,0 +1,180 @@
+"""
+def interpret(command_list, in_file, offset):
+    ret = ""
+    end = False
+    ok = True
+    for line in command_list:
+        offset += 1
+        if end:
+            print("%s:%d:0: error: Trailing instructions after 'end'"
+                  % (in_file, offset), file=sys.stderr)
+            exit(1)
+
+        m = re.match(r"^$", line)
+        if m:
+            continue
+
+        m = re.match(r"^//.*$", line)
+        if m:
+            continue
+
+        m = re.match(r"^execute line (?P<l>\d+)$", line)
+        if m:
+            ret += ("BMC at file[^\\n]+line %d function \\w+\\n"
+                    % int(m.group("l")))
+            continue
+
+        m = re.match(r"^save line (?P<l>\d+)$", line)
+        if m:
+            ret += ("Saving 'file[^\\n]+line %d function \\w+'\\n"
+                    % int(m.group("l")))
+            continue
+
+        m = re.match(r"^resume line (?P<l>\d+)$", line)
+        if m:
+            ret += ("Resuming from 'file[^\\n]+line %d function \\w+'\\n"
+                    % int(m.group("l")))
+            continue
+
+        m = re.match(r"^unwind$", line)
+        if m:
+            ret += "Unwinding loop [^\\n]+\\n"
+            continue
+
+        m = re.match(r"^path is unsuccessful$", line)
+        if m:
+            ret += "VERIFICATION FAILED\\n"
+            continue
+
+        m = re.match(r"^path is successful$", line)
+        if m:
+            ret += "VERIFICATION SUCCESSFUL\\n"
+            continue
+
+        m = re.match(r"^end$", line)
+        if m:
+            ret += "[^.]"
+            end = True
+            continue
+
+        m = re.match(r"^any lines$", line)
+        if m:
+            # Every other pattern deposits a '\n' at the end of the string.
+            # Erase those two characters and add a pattern to match multiple
+            # lines instead.
+            ret = "%s%s" % (ret[:-2], "(.|\\n)+")
+            continue
+
+        print("%s:%d:0: error: Syntax error in path exploration test. "
+              "(line: '%s')" % (in_file, offset, line), file=sys.stderr)
+        ok = False
+    if not ok:
+        exit(1)
+    return ret
+
+
+def main():
+    for dyr in os.listdir("."):
+        if not os.path.isdir(dyr):
+            continue
+        for fyle in os.listdir(dyr):
+            base, ext = os.path.splitext(fyle)
+            if ext != ".in":
+                continue
+            in_file = os.path.join(dyr, fyle)
+            sections = get_desc_sections(in_file)
+            if len(sections) == 4:
+                offset = (len(sections[0]) + len(sections[1]) +
+                          len(sections[2]) + 3)
+                sections[0].append(interpret(sections[3], in_file, offset))
+                sections[3] = [
+                    "             WARNING:",
+                    "This .desc file is automatically generated",
+                    "from the file '{fyle}'.".format(fyle=in_file),
+                    "Do not modify this file."
```

Comment:
This needs a `\n` at the end to get rid of my complaint about missing-newline-at-end-of-file.